File: example/go-redis/redis/sorted-set.go
package redis

import (
	"context"

	redisv9 "github.com/redis/go-redis/v9"
)

func (r *repository) ZADD(ctx context.Context, key string, score float64, member string) error {
	return r.client.ZAdd(
		ctx,
		key,
		redisv9.Z{
			Score:  score,
			Member: member,
		},
	).Err()
}

func (r *repository) ZREVRANGE(ctx context.Context, key string, start, stop int64) ([]string, error) {
	return r.client.ZRangeArgs(
		ctx,
		redisv9.ZRangeArgs{
			Key:   key,
			Start: start,
			Stop:  stop,
			Rev:   true,
		},
	).Result()
}

File: example/rust-redis/src/redis/sorted_set.rs
use anyhow::Result;
use redis::AsyncCommands;

use super::redis::Repository;

impl Repository {
    pub async fn zadd(&self,key: &str,score: f64,member: &str) -> Result<()> {
        let mut conn = self.client.get_multiplexed_async_connection().await?;
        conn.zadd::<_, _, _, ()>(
            key,
            member,
            score,
        )
        .await?;
        Ok(())
    }

    pub async fn zrevrange(&self,key: &str,start: isize,stop: isize) -> Result<Vec<String>> {
        let mut conn = self.client.get_multiplexed_async_connection().await?;
        let ranking: Vec<String> = conn
            .zrevrange(key, start, stop)
            .await?;

        Ok(ranking)
    }
}

